apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx

```diff
@@ -0,0 +1,136 @@
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import TextareaWithMarkdownPreview from '@cdo/apps/lib/levelbuilder/TextareaWithMarkdownPreview';
+import SaveBar from '@cdo/apps/lib/levelbuilder/SaveBar';
+import {navigateToHref} from '@cdo/apps/utils';
+import $ from 'jquery';
+import color from '@cdo/apps/util/color';
+
+export default function ProgrammingExpressionEditor({
+  initialProgrammingExpression
+}) {
+  //const [name, setName] = useState(initialProgrammingExpression.name);
+  //const [shortDescription, setShortDescription] = useState(
+  //  initialProgrammingExpression.shortDescription
+  //);
+  const [programmingExpression, setProgrammingExpression] = useState({
+    ...initialProgrammingExpression,
+    id: null,
+    key: null
+  });
```

Comment:
nit: i think it would be clearer to destructure `initialProgrammingExpression` to exclude the `id` and `key` keys like

```javascript
const {id, key, ...remainingProgrammingExpression} = initialProgrammingExpression;
const [programmingExpression, setProgrammingExpression] = useState(remainingProgrammingExpression);

// now we can also use id and key everywhere instead of initialProgrammingExpression[id|key]
```

`lastUpdated` is currently included in this state and in the state on line 22 below, so we should also de-dup that

Reply:
yeah I was trying to do this without the extra variable because I didn't want to destructure on every render but it wasn't working. I can change it though.

I'll dedup `lastUpdated`

Reply:
wait jk, how is `lastUpdated` included in this state?

Reply:
ah, `lastUpdated` isn't actually in this object, I just forgot to remove that initialization. Easy fix :)

Reply:
hm our linter doesn't like the destructuring as it adds unused variables. I think we could change the rules to set [ignoreRestSiblings](https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings) to `true` but I don't know if I want to do that in this PR. thoughts?

Reply:
i don't think we want to update the linting rules, but we're referencing `initialProgrammingExpression.id` (and `.key`) throughout this component -- we could replace those references with the destructured variable

Reply:
ah, makes sense. Thanks!